src/main/java/com/petfinder/service/UserService.java

```diff
@@ -0,0 +1,77 @@
+package com.petfinder.service;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+import org.apache.commons.validator.routines.EmailValidator;
+import org.mindrot.jbcrypt.BCrypt;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.petfinder.dao.UserRepository;
+import com.petfinder.domain.User;
+import com.petfinder.exception.EmailExistsException;
+import com.petfinder.exception.InvalidEmailException;
+import com.petfinder.exception.LoginExistsException;
+
+@Service
+public class UserService {
+
+	@Autowired
+	private UserRepository userRepository;
+
+	@Transactional
+	public void register(String login, String password, String repeatpassword, String email)
+			throws LoginExistsException, EmailExistsException, InvalidEmailException {
+		if (isEmailAddressValid(email)) {
+			if (verifyLogin(login) && verifyEmail(email)) {
+				if (ifPasswordsMatch(password, repeatpassword)) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://www.codacy.com/app/matikz1110/PetFinder/file/2508689008/issues/source?bid=3239464&fileBranchId=3276039#l29)